pkg/testcase/upgradenodereplacement.go

```diff
@@ -598,3 +598,25 @@ func buildRke2Cmd(
 
 	return cmd, nil
 }
+
+// DeleteReplacedNodes deletes the new nodes created as replacements.
+func DeleteReplacedNodes(cluster *shared.Cluster) {
+	externalIPs := shared.FetchNodeExternalIPs()
+	awsClient, err := aws.AddAWSClient(cluster)
+	Expect(err).NotTo(HaveOccurred(), "error creating aws client: %s", err)
+
+	var wg sync.WaitGroup
+	for _, ip := range externalIPs {
+		ip := ip
+		wg.Add(1)
+		go func(ip string) {
+			defer wg.Done()
+			nodeDelErr := awsClient.DeleteInstance(ip)
+			if nodeDelErr != nil {
+				shared.LogLevel("error", "on deleting node with ip: %v, got error %w", ip, nodeDelErr)
+				return
+			}
+		}(ip)
+		wg.Wait()
```

Comment:
the wait should be outside for loop, this way you are actually doing nothing

Reply:
FYI I copied the code from here which has similar wait inside the for loop... so this needs to be updated as well then? https://github.com/rancher/distros-test-framework/blob/main/entrypoint/rebootinstances/rebootinstances_suite_test.go#L74

Reply:
yes please, sorry about that , doing that way we are actually not using the async benefits

Reply:
if you dont mind updating, tks

Reply:
np will update